pycbc/dq.py

```diff
@@ -203,15 +203,18 @@ def query_flag(ifo, segment_name, start_time, end_time,
                     partial.append(segment(seg_start, seg_end))
 
                 # Limit to the veto definer stated valid region of this flag
-                send = segmentlist([segment([flag['start'],
-                                             flag['end']])])
+                send = segmentlist( [segment(flag['start'], flag['end'])] )
                 flag_segments += (partial.coalesce() & send)
 
-        else:  # Standard case just query directly.
+        else:  # Standard case just query directly
             try:
                 segs = query(':'.join([ifo, segment_name]),
                              int(start_time), int(end_time),
                              host=server)['active']
+                # dqsegdb output is not guaranteed to lie entirely within start
+                # and end times
+                segs = segs.coalesce() & segmentlist( [segment(int(start_time),
+                                                               int(end_time))] )
```

Comment:
It would be better to move this to line 224 at the same indent level as line 180 (and operating on `flag_segments`) so as to catch calls using veto-definer flags and standard flags.

However, this change will fix the problem, so I'll approve this now and treat the above as optional.